subprojects/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/ResourceFilter.java

```diff
@@ -0,0 +1,147 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.plugins.ide.eclipse.model;
+
+import com.google.common.base.Objects;
+import groovy.lang.Closure;
+import groovy.lang.GroovyObjectSupport;
+import org.gradle.util.ConfigureUtil;
+
+import java.util.UUID;
+
+/**
+ * The gradle DSL model of an Eclipse resource filter.
+ *
+ * Later this will be extended with additional helpers, but for now it just allows specifying a filter with a custom matcher
+ * and configuring whether it is an include/exclude filter that applies to files, folders, or both.  This example excludes the 'node_modules' folder.
+ *
+ * <pre autoTested=''>
+ * apply plugin: 'java'
+ * apply plugin: 'eclipse'
+ *
+ * eclipse {
+ *   project {
+ *     resourceFilters {
+ *       // the following closure configures the ResourceFilter
+ *       filter {
+ *         appliesTo = 'FOLDERS'
+ *         type = 'EXCLUDE_ALL'
+ *         matcher {
+ *           id = 'org.eclipse.ui.ide.multiFilter'
+ *           arguments = '1.0-name-matches-false-false-node_modules'
+ *         }
+ *       }
+ *     }
+ *   }
+ * }
+ * </pre>
+ *
+ * @since NEXT_VERSION
+ */
+public class ResourceFilter extends GroovyObjectSupport {
+    // Originally the spec mentioned using current epoch time, but all the filters tend to be created
+    // in the same closure and collisions were possible, so a random UUID was used instead.
+    private String id = UUID.randomUUID().toString();
+    private ResourceFilterAppliesTo appliesTo;
+    private ResourceFilterType type;
+    private Boolean recursive;
+    private ResourceFilterMatcher matcher;
+
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Sets the id of this resource filter.  If not set, the default id will be a random UUID.  As id is normally autogenerated, it is ignored by equals/hashCode.
+     */    
+    public void setId(String id) {
+        this.id = id;
+    }
+    
+    public ResourceFilterAppliesTo getAppliesTo() {
+        return appliesTo;
+    }
+    
+    public void setAppliesTo(ResourceFilterAppliesTo appliesTo) {
+        this.appliesTo = appliesTo;
+    }
+    
+    public ResourceFilterType getType() {
+        return type;
+    }
+
+    public void setType(ResourceFilterType type) {
+        this.type = type;
+    }
+
+    public Boolean getRecursive() {
+        return recursive;
+    }
+
+    public void setRecursive(Boolean recursive) {
+        this.recursive = recursive;
+    }
+
+    public ResourceFilterMatcher getMatcher() {
+        return matcher;
+    }
+
+    public void setMatcher(ResourceFilterMatcher matcher) {
+        this.matcher = matcher;
+    }
+
+    public void matcher(Closure configureClosure) {
```

Comment:
Ok, I see some other examples of using both an Action and a ClosureBackedAction in the code, I'll do it like that.  Should I also add an Action overload for EclipseProject.resourceFilter?